modules/apps/captcha/captcha-taglib/src/main/resources/META-INF/resources/captcha/simplecaptcha.jsp

```diff
@@ -39,23 +39,50 @@ String url = (String)request.getAttribute("liferay-captcha:captcha:url");
 
 			if (refreshCaptcha && !hasEventAttached) {
 				hasEventAttached = true;
-				refreshCaptcha.addEventListener('click', () => {
-					var url = Liferay.Util.addParams(
-						't=' + Date.now(),
-						'<%= HtmlUtil.escapeJS(url) %>'
-					);
-
-					var captcha = document.getElementById(
-						'<portlet:namespace />captcha'
-					);
-
-					if (captcha) {
-						captcha.setAttribute('src', url);
-					}
-				});
+				refreshCaptcha.addEventListener('click', handleCaptcha);
 			}
 		}
 
+		function handleCaptcha() {
+			var captcha = document.getElementById(
+				'<portlet:namespace />captcha'
+			);
+
+			if (<%= url.startsWith("/o/") %>) {
+				return getCaptchaSourceImage('<%= url %>', captcha);
+			}
+
+			let url = Liferay.Util.addParams(
+				't=' + Date.now(),
+				'<%= HtmlUtil.escapeJS(url) %>'
+			);
+
+			if(captcha) {
+				captcha.setAttribute('src', url);
+			}
+		}
+
+		function getCaptchaSourceImage(url, captcha) {
+			Liferay.Util.fetch(url, {
+				method: 'GET',
+			})
+			.then((response) => {
+				if (!response.ok) {
+					throw new Error();
+				}
+
+				return response.json();
+			})
+			.then((response) => {
+				captcha.setAttribute('src', response.image);
```

Comment:
Will get an issue here when `captcha` is null. Please double-check.

Reply:
Fixed it. Thanks